src/components/course/data/hooks.jsx

```diff
@@ -443,7 +443,7 @@ export const useTrackSearchConversionClickHandler = ({ href = undefined, eventNa
  * @returns Click handler function for clicks on enrollment buttons.
  */
 export const useOptimizelyEnrollmentClickHandler = ({ href, courseRunKey, userEnrollments }) => {
-  const hasNoExistingEnrollments = userEnrollments?.length === 0 || true;
+  const hasNoExistingEnrollments = !userEnrollments || userEnrollments.length === 0;
```

Comment:
[inform] Refactor based on the question/bug reported by Troy in #enterprise-titans Slack channel yesterday.